src/tools/ToolsCata/map_extractor/System.cpp

```diff
@@ -326,7 +325,7 @@ void ReadAreaTableDBC()
     }
 
     size_t area_count = dbc.getRecordCount();
-    maxAreaId = dbc.getMaxId();
+    size_t maxAreaId = dbc.getMaxId();
```

Comment:
maxAreaId is now a local var hiding the global var (uint32). It is needed for the areaflags
Line 488 (... areaid <= maxAreaId) is now always false and the area_flags never gets properly set.

Did you test these changes?